examples/dmd/de_dg_advection_greedy.cpp

```diff
@@ -0,0 +1,1635 @@
+    args.AddOption(&greedy_subset_size, "-greedysubsize", "--greedysubsize",
+                   "The greedy algorithm subset size.");
+    args.AddOption(&greedy_convergence_subset_size, "-greedyconvsize",
+                   "--greedyconvsize", "The greedy algorithm convergence subset size.");
+// Differential Evolution Parameters
+    args.AddOption(&de, "-de", "--de",
+                   "-no-de", "--no-de",
+                   "Enable or disable the differential evolution phase.");
+    args.AddOption(&de_min_f_factor, "-de_min_ff", "--de_min_f-factor",
+                   "DE min f-factor");
+    args.AddOption(&de_max_f_factor, "-de_max_ff", "--de_max_f-factor",
+                   "DE max f-factor");
+    args.AddOption(&de_F, "-de_f", "--de_f",
+                   "DE F.");
+    args.AddOption(&de_CR, "-de_cr", "--de_cr",
+                   "DE CR.");
+    args.AddOption(&de_PS, "-de_ps", "--de_ps",
+                   "DE Population size.");
+    args.AddOption(&de_min_iter, "-de_min_iter", "--de_min_iter",
+                   "DE Minimum number of iterations.");
+    args.AddOption(&de_max_iter, "-de_max_iter", "--de_max_iter",
+                   "DE Maximum number of iterations.");
+    args.AddOption(&de_ct, "-de_ct", "--de_ct",
+                   "DE Cost threshold.");
+    args.Parse();
+    if (!args.Good())
+    {
+        if (mpi.Root())
+        {
+            args.PrintUsage(cout);
+        }
+        return 1;
+    }
+    if (mpi.Root())
+    {
+        args.PrintOptions(cout);
+    }
+
+    Device device(device_config);
+    if (mpi.Root()) {
+        device.Print();
+    }
+
+    // 3. Initialize the DMD database that will be built using a greedy algorithm.
+    if(de)
+    {
+        target_f_factor = f_factor;
+
+        simulation();
+
+        // Create relative error cost function in 4 dimensions (radius, alpha, cx, cy)
+        RelativeDifferenceCostFunction cost(1);
+
+        // Create Differential Evolution optimizer with population size of de_ps
+        // differential weight of de_F, and crossover probability of de_CR
+        CAROM::DifferentialEvolution de_opt(cost, de_PS, de_F, de_CR);
+
+        // Optimize for at least de_min_iter iterations, to a maximum of de_max_iter iterations with verbose output.
+        // Stop early, after de_min_iter iterations is run, if the minimum cost did not improve by de_ct
+        std::vector<double> optimal_parameters = de_opt.Optimize(de_min_iter,
+                de_max_iter, de_ct, true);
+
+        f_factor = optimal_parameters[0];
+
+        simulation();
+
+        delete true_solution_u;
+
+        MPI_Finalize();
+
+        return 0;
+    }
+    else if (build_database)
+    {
+        MFEM_VERIFY(rdim != -1, "rdim must be set.");
+        MFEM_VERIFY(!visit
+                    && !visualization,
+                    "visit and visualization must be turned off during the build_database phase.")
+        std::ifstream infile("dg_advection_greedy_data");
```

Comment:
There are many files written to the current directory, which are cumbersome to delete. It would be more convenient to create a subdirectory and write the files there.

Reply:
I haven't done this yet.  I don't necessarily disagree with the statement, I just have to touch the it in several places to make this work, and I don't have time set aside to do so yet.

Reply:
I agree with @dylan-copeland - putting a lot of these files in a subdirectory would be helpful. This same problem exists across almost every other example in libROM though, so maybe this could be done in a separate PR that makes this change for all of the examples?

Reply:
Other examples do have this issue as well, but this one stands out as having a very large number of files. I am fine with fixing it just for this example or planning a future PR to do this for all examples.

Reply:
I have already talked with @ptranq, but let's make subfolder for all the output files at least for this example in this PR. We can gradually fixing other examples later.

Reply:
I have addressed this issue.  The code is now output into a default directory, and this directory can be modified via a command line argument.  Additionally, an input directory can be specified to identify where previously written files are located.

Reply:
Further changed to use a single io directory, since some stages wanted to read and write in the same call.